server/oauth.ts

```diff
@@ -346,12 +325,14 @@ export const setLocalStateFromIdTokenOrUserCookie = (
 	// but not the other fields. This will allow the frontend to show the
 	// signed in menu, but not show the user's name or email.
 	const hasIdTokenOrUserCookie = idToken || req.cookies['GU_U'];
-	res.locals.identity = {
+
+	const result = IdTokenClaims.safeParse(idToken?.claims);
+	setIdentityLocalState(res, {
 		signInStatus: hasIdTokenOrUserCookie ? 'signedInRecently' : undefined,
-		userId: idToken?.claims.legacy_identity_id,
-		name: idToken?.claims.name,
-		email: idToken?.claims.email,
-	};
+		userId: result.success ? result.data.legacy_identity_id : undefined,
+		displayName: result.success ? result.data.name : undefined,
+		email: result.success ? result.data.email : undefined,
+	});
```

Comment:
@coldlink I changed this after your review so that Zod doesn't throw when the `idToken` isn't set, and `setIdentityLocalState()` is called with the 'best possible' data - either the result of the Zod parse, or `undefined`.